source/core/Waveform.cpp

```diff
@@ -0,0 +1,54 @@
+/**
+
+Copyright (c) 2014-2015 National Instruments Corp.
+
+This software is subject to the terms described in the LICENSE.TXT file
+
+SDG
+*/
+
+/*! \file
+\brief Analog waveform functions
+*/
+
+#include "TypeDefiner.h"
+#include "ExecutionContext.h"
+#include "TypeAndDataManager.h"
+#include "VirtualInstrument.h"
+#include <vector>
+#include <cmath>
+#include "Waveform.h"
+
+using namespace Vireo;
+
+#ifdef VIREO_TYPE_Waveform
+DEFINE_VIREO_BEGIN(Waveform)
+DEFINE_VIREO_REQUIRE(IEEE754Math)
+DEFINE_VIREO_REQUIRE(Timestamp)
+DEFINE_VIREO_TYPE(AnalogWaveform, "c( e(Timestamp t0) e(Double dt) e(a($0 $1)Y) )")
+DEFINE_VIREO_TYPE(DigitalWaveform, "c(e(a(UInt8 * *) data)e(a(UInt32 *) transitions))")
+DEFINE_VIREO_END()
+#endif
+
+VIREO_FUNCTION_SIGNATURE4(AnalogWaveformBuild, AnalogWaveform, Timestamp, Double, TypedObjectRef)
+{
+    _Param(0)._t0 = _Param(1);
+    _Param(0)._dt = _Param(2);
+
+    TypedObjectRef* argY_source = _ParamPointer(3);
```

Comment:
Probably use TypedArrayCoreRef* for these type (and for the last func arg) to more clearly indicate it's an array